### PR TITLE
Feature/321 apply to offer permissions

### PR DESCRIPTION
**Pull request by BlancoSebastianEzequiel**

### CODE REVIEW

File: src/models/JobApplication/Errors/OfferNotTargetedForApplicantError.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { ApplicantType } from "$models/Offer";
+
+export class OfferNotTargetedForApplicantError extends Error {
+  public static buildMessage(applicantType: ApplicantType, targetApplicantType: ApplicantType) {
+    return `Applicant of type ${applicantType} cannot apply to offer targeted for ${targetApplicantType}`;
```
Comment (dylanalvarez) [inline src/models/JobApplication/Errors/OfferNotTargetedForApplicantError.ts:5]: no es "targeted for students" es "targeted to students"

File: src/models/Offer/Model.ts [hunk 2]
```diff
@@ -119,4 +119,11 @@ export class Offer extends Model<Offer> {
   public getSections: HasManyGetAssociationsMixin<OfferSection>;
   public getCareers: HasManyGetAssociationsMixin<Career>;
   public getApprovalEvents: HasManyGetAssociationsMixin<OfferApprovalEvent>;
+
+  public async applicantCanApply(applicant: Applicant) {
+    const applicantType = await applicant.getType();
+    if (this.targetApplicantType === ApplicantType.both) return true;
```
Comment (dylanalvarez) [inline src/models/Offer/Model.ts:125]: en este caso se hace la query al pedo, lo pondria arriba

**Review by dylanalvarez — APPROVED** [review]

File: src/models/JobApplication/Errors/OfferNotTargetedForApplicantError.ts (hunk 1 above)
Comment (llmanuel) [reply src/models/JobApplication/Errors/OfferNotTargetedForApplicantError.ts:5]: `A ${applicantType} cannot apply to offer for ${targetApplicantType}`

File: test/models/Offer/Repository.test.ts [hunk 3]
```diff
@@ -177,6 +179,79 @@ describe("OfferRepository", () => {
...
+        const { uuid } = await OfferRepository.create(
+          OfferGenerator.data.withObligatoryData({
+            companyUuid,
+            targetApplicantType
+          })
+        );
+        return await OfferRepository.updateApprovalStatus({ uuid, status, admin });
+      };
+
+      it("returns true if the offer targets students and the applicant is a student", async () => {
```
Comment (llmanuel) [inline test/models/Offer/Repository.test.ts:201]: returns true if the offer is for students and the applicant is a student

File: test/models/Offer/Repository.test.ts [hunk 4]
```diff
@@ -177,6 +179,79 @@ describe("OfferRepository", () => {
...
+        const offer = await createOfferFor(ApplicantType.student, ApprovalStatus.approved);
+        const applicant = await ApplicantGenerator.instance.student();
+        expect(await offer.applicantCanApply(applicant)).toBe(true);
+      });
+
+      it("returns true if the offer targets graduates and the applicant is a graduate", async () => {
```
Comment (llmanuel) [inline test/models/Offer/Repository.test.ts:207]: idem el resto

**Review by llmanuel — APPROVED** [review]

**Pull request merged**